pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -351,19 +341,43 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env model.En
 				}
 
 				listenerOpts.protocol = model.ProtocolHTTP
-				listenerOpts.httpOpts = &httpListenerOpts{
-					//rds:              fmt.Sprintf("%d", servicePort.Port),
-					routeConfig: configgen.BuildSidecarOutboundHTTPRouteConfig(env, node, proxyInstances, services,
-						fmt.Sprintf("%d", servicePort.Port)),
-					useRemoteAddress: useRemoteAddress,
-					direction:        operation,
+				listenerOpts.filterChainOpts = []*filterChainOpts{{
+					httpOpts: &httpListenerOpts{
+						//rds:              fmt.Sprintf("%d", servicePort.Port),
+						routeConfig: configgen.BuildSidecarOutboundHTTPRouteConfig(
+							env, node, proxyInstances, services, fmt.Sprintf("%d", servicePort.Port)),
+						useRemoteAddress: useRemoteAddress,
+						direction:        operation,
+					},
+				}}
+			default:
+				log.Infof("buildSidecarOutboundListeners: service %q has unknown protocol %#v, defaulting to TCP", service.Hostname, servicePort)
+				fallthrough
+			case model.ProtocolTCP, model.ProtocolHTTPS, model.ProtocolMongo, model.ProtocolRedis:
+				listenerType = plugin.ListenerTypeTCP
```

Comment:
if the rules for setting listener type are not trivial, it may be better to remove the helper in the other instance of this. otherwise why couldn't we use the same function to set listener type here?

Reply:
Unfortunately the cases don't collapse together in quite the same way as the Gateway case.

Updated to use the help method but did not collapse the similar logic in the cases (I'd prefer to do that in a separate PR).